yt/units/yt_array.py

```diff
@@ -728,6 +728,8 @@ def to_equivalent(self, unit, equiv, **kwargs):
         >>> a.to_equivalent("keV", "thermal")
         """
         conv_unit = Unit(unit, registry=self.units.registry)
+        if self.units.same_dimensions_as(conv_unit):
```

Comment:
@ngoldbaum let me know what you think of this. It's supposed to catch something silly like this:
```python
kT = yt.YTQuantity(3.0, "keV")
kT.in_units("erg", equivalence="thermal")
```